### ISSUE TITLE
Discrepancy between Manuscript Formula (2) and code for MGAE

### ISSUE THREAD

**Issue by sunyinhe114** [ZJUFanLab/scNiche#25]
Hi team,

I am studying the implementation of scniche and noticed a significant inconsistency between the mathematical description in the manuscript and the source code in scnishe/trainer/_train.py, line 72. 

In the manuscript "https://doi.org/10.1038/s41467-025-57029-9", Formula (2) suggests that the joint representation Z(2) is calculated by applying a graph convolution during the fusion step: 
$$Z_{(2)} = \delta \left( \sum_{v=1}^{V} W_a^{(v)} \left( (\tilde{D}^{(v)})^{-\frac{1}{2}} \tilde{A}^{(v)} (\tilde{D}^{(v)})^{-\frac{1}{2}} Z_{(1)}^{(v)} W_{(2)}^{(v)} \right) \right)$$

However, in the FeatureFusion class, the implementation is a weighted sum followed by a Softmax and Activation: 
"def forward(self, *features):
        z = sum(F.dropout(f, self.dropout) @ w for f, w in zip(features, self.weights))
        z = F.softmax(z, dim=1)
        z = self.activation(z)
        return z", the formula for this should be : $$Z_{\text{(2)}} = \delta \left( \text{Softmax} \left( \sum_{v=1}^{V} \text{Dropout}(Z_{(1)}^{(v)}) W_a^{(v)} \right) \right)$$

Question: Is the implementation in the code the intended "optimized" version of the model, and should the manuscript formulas be interpreted as simplified theoretical representations? Thank you for taking your time! 

Best regards, Yinhe

**Comment by ProDong0512** [comment]
> Hi team,
> 
> I am studying the implementation of scniche and noticed a significant inconsistency between the mathematical description in the manuscript and the source code in scnishe/trainer/_train.py, line 72.
> 
> In the manuscript "https://doi.org/10.1038/s41467-025-57029-9", Formula (2) suggests that the joint representation Z(2) is calculated by applying a graph convolution during the fusion step: Z ( 2 ) = δ ( ∑ v = 1 V W a ( v ) ( ( D ~ ( v ) ) − 1 2 A ~ ( v ) ( D ~ ( v ) ) − 1 2 Z ( 1 ) ( v ) W ( 2 ) ( v ) ) )
> 
> However, in the FeatureFusion class, the implementation is a weighted sum followed by a Softmax and Activation: "def forward(self, *features): z = sum(F.dropout(f, self.dropout) @ w for f, w in zip(features, self.weights)) z = F.softmax(z, dim=1) z = self.activation(z) return z", the formula for this should be : Z (2) = δ ( Softmax ( ∑ v = 1 V Dropout ( Z ( 1 ) ( v ) ) W a ( v ) ) )
> 
> Question: Is the implementation in the code the intended "optimized" version of the model, and should the manuscript formulas be interpreted as simplified theoretical representations? Thank you for taking your time!
> 
> Best regards, Yinhe

Thanks for your concern! Your comprehension is correct. The code implementation reflects an optimized and modular software design where these two steps—graph convolution and feature fusion—are separated for clarity and efficiency. And in practical implementation, the graph convolution code is in model.py, line 71, executed before feature_fusion step. Hope this would clear your doubt!

**Comment by sunyinhe114** [comment]
Hi authors,

Thank you for your explanation! I want to consult another question. While reviewing the implementation of the Multi-view Graph Autoencoder (MGAE), I noticed a potential mismatch between the mathematical formulation in the paper and the provided Python code regarding the view-specific decoders. 

Mathematical FormulationAccording to the paper in Formula (4), the decoder is defined as:
$$\hat{A}^{(v)} = \text{sigmoid}(Z \cdot W^{(v)} \cdot Z^T)$$

However, in the provided code, the InnerProductDecoder class is implemented as follows:
```python
class InnerProductDecoder(nn.Module):
    def __init__(self, activation=torch.sigmoid, dropout=0.1):
        super(InnerProductDecoder, self).__init__()
        self.dropout = dropout
        self.activation = activation

    def forward(self, z):
        z = F.dropout(z, self.dropout)
        # This calculates Z * Z^T
        adj = self.activation(torch.mm(z, z.t())) 
        return adj
```
And in the MGAE forward pass:
```python
adj_rec = {i: self.decoder(feat_fusion) for i in range(self.views)}
```
This means this decoder calculates $$\(Z \cdot Z^T)$$ rather than $$\(Z \cdot W^{(v)} \cdot Z^T)$$

Concerns
1. Identical Reconstructions: Because the InnerProductDecoder lacks the $W^{(v)}$ matrix, adj_rec[0], adj_rec[1], etc., will result in identical values. This seems to bypass the "view-specific" reconstruction logic mentioned in the text.

2. Missing Parameters: The learnable parameters $W^{(v)}$ defined in the paper do not appear to be present in this part of the implementation.

Questions
Was the omission of $W^{(v)}$ intentional for simplification or performance reasons?

I would appreciate your clarification. Thank you very much!
